cache/demux.go

```diff
@@ -188,10 +219,8 @@ func (d *demux) resyncLaggingWatchers() {
 			d.laggingWatchers[w] = nextRev
 		}
 	}
-}
-
-func (d *demux) PeekOldest() int64 {
-	d.mu.RLock()
-	defer d.mu.RUnlock()
-	return d.history.PeekOldest()
+	d.mu.Unlock()
+	for _, c := range compacted {
+		c.w.notifyCompacted(c.rev)
```

Comment:
This is a blocking operation. While moving it outside of lock helps unlock other operation, but means that resync loop would be stuck deadlocked if any watch would not read response.